api/api.py

```diff
@@ -174,6 +176,45 @@ def post(self):
         return {}, 200
 
 
+class APDU:
+    def __init__(self):
+        # We want to be notified when APDU response is transmitted from the SE
+        self.condition = threading.Condition()
+
+    def exchange(self, data: bytes) -> bytes:
+        with self.condition:  # Lock for a command/response for one client
+            self.response = None
+            seph.to_app(data)
+            with self.condition:
+                while self.response is None:
+                    self.condition.wait()
+            return bytes(self.response)  # clone before leaving lock
+
+    def seph_apdu_callback(self, data: bytes):
+        """
+        Called by seph when data is transmitted by the SE. That data should
+        be the response to a prior APDU request
+        """
+        self.response = data
```

Comment:
* Maybe this code should also lock `self.condition` before setting `self.response`
* If `self.response` is not `None`, this means that no one is expecting the response or that a response was already received and will be overwritten. Showing a warning in Speculos console in such a case would help debug unexpected issues. What do you think?

Reply:
Fixed